### ISSUE TITLE
cmd process too long.

### ISSUE THREAD

**Issue by YiChenCityU** [youngLBW/HRN#3]
Hi, thanks for your contribution.  I test the project, but got stuck as follows: . It takes very long time. Do you know why?

CUDA_VISIBLE_DEVICES=0 python demo.py --input_type single_view --input_root ./assets/examples/single_view_image --output_root ./assets/examples/single_view_image_results
2023-04-19 21:23:19.692314: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library libcudart.so.10.1
/home/eason/anaconda3/envs/HRN/lib/python3.6/site-packages/tensorflow/python/autograph/utils/testing.py:21: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
----------------- Options ---------------
                add_image: True                          
           base_ckpt_path:                               
               bfm_folder: assets/3dmm_assets/BFM        
                bfm_model: BFM_model_front.mat           
                 camera_d: 10.0                          
                   center: 112.0                         
          checkpoints_dir: assets/pretrained_models      
             dataset_mode: None                          
                 ddp_port: 12355                         
        display_per_batch: True                          
                    epoch: 10                            
          eval_batch_nums: inf                           
                    focal: 1015.0                        
                  gpu_ids: 0                             
               img_folder: examples                      
                init_path: checkpoints/init_model/resnet50-0676ba61.pth
                  isTrain: False                         	[default: None]
                    model: facerecon                     
                     name: hrn_v1.1                      
                net_recog: r50                           
           net_recog_path: ../pretrained_models/recog_model/ms1mv3_arcface_r50_fp16/backbone.pth
                net_recon: resnet50                      
                    phase: test                          
                rot_angle: 10.0                          
              scale_delta: 0.1                           
               shift_pixs: 10.0                          
                   suffix:                               
            use_crop_face: True                          
                  use_ddp: False                         	[default: True]
              use_last_fc: False                         
             use_predef_M: False                         
                  verbose: False                         
           vis_batch_nums: 1                             
                    w_adv: 1.0                           
                  w_color: 1.92                          
                w_contour: 20.0                          
                w_dis_reg: 10.0                          
                    w_exp: 0.8                           
                   w_feat: 0.2                           
                  w_gamma: 10.0                          
                     w_id: 1.0                           
                     w_lm: 0.0016                        
                  w_reflc: 5.0                           
                    w_reg: 0.0003                        
                 w_smooth: 5000.0                        
                    w_tex: 0.017                         
               world_size: 1                             
                    z_far: 15.0                          
                   z_near: 5.0                           
----------------- End -------------------

environment： 
Package                 Version
----------------------- ------------------
absl-py                 1.4.0
albumentations          1.3.0
astunparse              1.6.3
cachetools              4.2.4
certifi                 2021.5.30
charset-normalizer      2.0.12
colorama                0.4.5
cycler                  0.11.0
dataclasses             0.8
decorator               4.4.2
einops                  0.4.1
face-alignment          1.3.5
future                  0.18.3
fvcore                  0.1.5.post20210915
gast                    0.3.3
google-auth             2.17.3
google-auth-oauthlib    0.4.6
google-pasta            0.2.0
grpcio                  1.48.2
h5py                    2.10.0
idna                    3.4
imageio                 2.15.0
importlib-metadata      4.8.3
importlib-resources     5.4.0
iopath                  0.1.9
joblib                  1.1.1
Keras-Preprocessing     1.1.2
kiwisolver              1.3.1
kornia                  0.5.11
llvmlite                0.36.0
Markdown                3.3.7
matplotlib              3.3.4
mkl-fft                 1.3.0
mkl-random              1.1.1
mkl-service             2.3.0
networkx                2.5.1
ninja                   1.11.1
numba                   0.53.1
numpy                   1.18.1
nvdiffrast              0.3.0
oauthlib                3.2.2
olefile                 0.46
opencv-python           4.5.5.64
opencv-python-headless  4.5.5.64
opt-einsum              3.3.0
packaging               21.3
Pillow                  8.4.0
pip                     21.2.2
portalocker             2.3.2
protobuf                3.19.6
pyasn1                  0.4.8
pyasn1-modules          0.2.8
pyparsing               3.0.9
python-dateutil         2.8.2
pytorch3d               0.6.1
PyWavelets              1.1.1
PyYAML                  6.0
qudida                  0.0.4
requests                2.27.1
requests-oauthlib       1.3.1
rsa                     4.9
scikit-image            0.17.2
scikit-learn            0.24.2
scipy                   1.4.1
setuptools              58.0.4
six                     1.16.0
tabulate                0.8.10
tensorboard             2.10.1
tensorboard-data-server 0.6.1
tensorboard-plugin-wit  1.8.1
tensorflow-estimator    2.3.0
tensorflow-gpu          2.3.0
termcolor               1.1.0
threadpoolctl           3.1.0
tifffile                2020.9.3
torch                   1.6.0
torchsummary            1.5.1
torchvision             0.7.0
tqdm                    4.64.1
trimesh                 3.21.5
typing_extensions       4.1.1
urllib3                 1.26.15
Werkzeug                2.0.3
wheel                   0.37.1
wrapt                   1.15.0
yacs                    0.1.8
zipp                    3.6.0

**Comment by youngLBW** [comment]
Hi, I haven't met this situation before. You may need to debug to find the precise location where the procedure gets stuck. I guess it may be caused by the environmental problem or the downloading of pre-trained models (face-alignment).
